### PR TITLE
grt: repair_antennas fixes

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -465,16 +465,22 @@ void GlobalRouter::restoreCapacities(Capacities capacities,
     for (int y = 1; y < yGrids; y++) {
       for (int x = 1; x < xGrids; x++) {
         oldCap = h_caps[layer - 1][y - 1][x - 1];
-        _fastRoute->addAdjustment(
-            x - 1, y - 1, layer, x, y - 1, layer, oldCap, true);
+        currCap = _fastRoute->getEdgeCapacity(x - 1, y - 1, layer, x, y - 1, layer);
+        if (oldCap <= currCap) {
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:469]: Would you explain what this is for?

File: src/grt/src/AntennaRepair.cpp [hunk 2]
```diff
@@ -247,7 +248,8 @@ void AntennaRepair::insertDiode(odb::dbNet* net,
   int leftPad = _opendp->padGlobalLeft();
   int rightPad = _opendp->padGlobalRight();
   std::vector<value> overlapInsts;
-  while (!legallyPlaced) {
+  int legalize_itr = 0;
+  while (!legallyPlaced && legalize_itr < max_legalize_itr) {
```
Comment (maliberty) [inline src/grt/src/AntennaRepair.cpp:252]: Should there be a warning if you hit max iters?

File: src/grt/src/GlobalRouter.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.cpp:474]: I've added a comment in the code trying to explain it.

File: src/grt/src/GlobalRouter.cpp [hunk 3]
```diff
@@ -454,7 +454,12 @@ void GlobalRouter::restoreCapacities(Capacities capacities,
                                      int previousMinLayer,
                                      int previousMaxLayer)
 {
-  int oldCap, currCap;
+  int oldCap;
+  // check if current edge capacity is larger than the old edge capacity
+  // before applying adjustments.
+  // after inserting diodes, edges can have less capacity than before,
+  // and apply adjustment without a check leads to warns and wrong adjustments.
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:461]: Thanks.  Please capitalize Chech and After at the start of sentences.

**Review by maliberty — APPROVED** [review]

**Pull request merged**